nrf_802154/driver/include/nrf_802154.h

```diff
@@ -102,32 +102,36 @@ void nrf_802154_deinit(void);
  */
 extern void nrf_802154_custom_part_of_radio_init(void);
 
-#if !NRF_802154_INTERNAL_RADIO_IRQ_HANDLING
+#if DOXYGEN || !NRF_802154_INTERNAL_RADIO_IRQ_HANDLING
+
 /**
  * @brief Handles the interrupt request from the RADIO peripheral.
  *
- * @note If NRF_802154_INTERNAL_RADIO_IRQ_HANDLING is enabled, the driver internally handles the
+ * @note If @ref NRF_802154_INTERNAL_RADIO_IRQ_HANDLING is enabled, the driver internally handles the
  *       RADIO IRQ, and this function must not be called.
  *
  * This function is intended for use in an operating system environment, where the OS handles IRQ
  * and indirectly passes it to the driver, or with a RAAL implementation that indirectly passes
  * radio IRQ to the driver (that is, SoftDevice).
  */
 void nrf_802154_radio_irq_handler(void);
+
 #endif // !NRF_802154_INTERNAL_RADIO_IRQ_HANDLING
 
-#if !NRF_802154_INTERNAL_SWI_IRQ_HANDLING
+#if DOXYGEN || !NRF_802154_INTERNAL_SWI_IRQ_HANDLING
```

Comment:
`#if` on certain symbols didn't work correctly because in doxyfile they were defined to empty symbols and DOXYGEN symbol was defined in not a separate line see, changes do nrfxlib.doxyfile.in in https://github.com/nrfconnect/sdk-nrf/pull/5622/files
However I will change to defined(DOXYGEN) and place them at the end as you suggest.